### PR TITLE
update rsyslog fixture to workaround /var/log size limitation

### PR DESCRIPTION
**Pull request by lipxu**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
The case failed sometimes due to cannot found the log.
It should be related to the small storage, since the var/log size limitation, logrotate would remove the previous syslogs during the test, then cause loganalyzer cannot find the previous log and teardown failure.

#### How did you do it?
It is caused by storage resource limitation, could not fix the issue, this PR is a workaround.
Since set the syslog to error lever and update the logroate to compress log immediately during the test still failed sometimes. Create a thread to do the logrotate every 60s during the test.

#### How did you verify/test it?
https://elastictest.org/scheduler/testplan/66d1ebb7f90c6cd947f4dd3e

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -2455,3 +2459,39 @@ def format_failure(port, failure):
...
+def run_logrotate(duthost, stop_event):
+    logger.info("Start rotate_syslog on {}".format(duthost))
+    while not stop_event.is_set():
+        try:
+            # Run logrotate for rsyslog
+            duthost.shell("logrotate -f /etc/logrotate.conf", module_ignore_errors=True)
+        except subprocess.CalledProcessError as e:
+            logger.error("Error: {}".format(str(e)))
+        # Wait for 60 seconds before the next rotation
+        time.sleep(60)
```
Comment (StormLiangMS) [inline tests/conftest.py:2473]: do we want 30 seconds or 60 seconds for each iteration?

Comment (lipxu) [reply tests/conftest.py:2473]: Thanks for review, updated the description, should be 60s.
based on the local test, the swss/sairedis.rec and swss/swss.rec log size increased very fast during the test, and seems "delaycompress" not working well, so I created a thread to rotate the log every 30s first, based on test, 60s should also OK, so I modify the sleep value to 60s.

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Comment by lipxu** [comment]
> @lipxu PR conflicts with 202311 branch

manually created the PR to resolve conflict, https://github.com/sonic-net/sonic-mgmt/pull/14380, @StormLiangMS , please help to review and merge, thanks a lot.

**Comment by lipxu** [comment]
@liushilongbuaa The original auto-created backport PR #14379 was closed due to conflicts. I have manually cherry-picked and resolved the conflicts — the 202405 backport is now available at #23221. Thank you for the heads-up!

**Pull request merged**